src/Tasks/Microsoft.NET.Build.Tasks/buildCrossTargeting/Microsoft.NET.Sdk.targets

```diff
@@ -31,4 +31,16 @@ Copyright (c) .NET Foundation. All rights reserved.
     <Error Text="The 'Publish' target is not supported without specifying a target framework. The current project targets multiple frameworks, please specify the framework for the published application." />
   </Target>
   
+  <!-- 
+    Nuget doesn't see or use runtime identifiers in restore that are only
+    conditionally used in some inner builds. 
+    
+    However, in the case where we're building .NETFramework exe on Windows 
+    (see *.RuntimeIdentifierInference.targets for more information) in an inner
+    build, we may implicitly use one of these RIDs and so we must ensure that restore 
+    includes them.
+   -->
+  <PropertyGroup Condition="'$(RuntimeIdentifiers)' == '' and '$(OS)' == 'Windows_NT'">
```

Comment:
Could we check to see whether `RuntimeIdentifiers` contains a `net*` target framework?  Even just checking if the string contains `net` would allow us to avoid this in most cases where we don't need it.